### PR TITLE
Aliquot covaris microbial

### PR DESCRIPTION
**Pull request by barrystokman**
### Added
New EPP to calculate microbial aliquot volumes. This EPP is used in the Master Step CG002 - Normalization of microbial samples, button Calculate Aliquot Volume in the workflow Nextera Microbial WGS v4.
Replaces the script https://github.com/Clinical-Genomics/clinical_EPPs/blob/master/EPPs/aliquot_covaris_microbial.py

**Steps to consider while deploying**
- Configuration changes:
- Documentation updates:
- Inform users by email:

### Review:
- [ ] Code approved by
- [ ] Tests executed on stage by:  (Document the test done with screen shots and description.)
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [x] **MINOR** - when you add functionality in a backwards compatible manner

### CODE REVIEW

**Comment by barrystokman** [comment]
## Positive testing
### Current EPP results:
image

**Comment by barrystokman** [comment]
## Positive testing
### Current EPP results:
image
### New EPP results:
image

## Negative testing:
### Missing concentration:
image
### Too high concentration:

**Comment by barrystokman** [comment]
## Positive testing
### Current EPP results:
image
### New EPP results:
image
## Negative testing
### Missing concentration:
image

File: cg_lims/EPPs/udf/calculate/calculate_microbial_aliquot_volumes.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from cg_lims.exceptions import HighConcentrationError, LimsError, MissingUDFsError
+from cg_lims.get.artifacts import get_artifacts
+
+FINAL_CONCENTRATION = 2
+LOG = logging.getLogger(__name__)
+MAXIMUM_CONCENTRATION = 60
+
+
+def calculate_volumes(artifacts: List[Artifact]) -> Optional[str]:
+    """Determines the total volume, water volume, and sample volume"""
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/calculate/calculate_microbial_aliquot_volumes.py:21]: Would be nice with reference to the method document, or some deceription of the different hardcoded rules. You know..

File: cg_lims/EPPs/udf/calculate/calculate_microbial_aliquot_volumes.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        artifact.udf["Total Volume (uL)"] = buffer_volume + sample_volume
+        artifact.put()
+
+    if missing_udfs:
+        raise MissingUDFsError(
+            f"Could not apply calculations for {missing_udfs} out of {len(artifacts)} sample(s): "
+            f"'Concentration' is missing!"
+        )
+
+    if high_concentration:
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/calculate/calculate_microbial_aliquot_volumes.py:63]: I think we should raise here as well.

**Review by mayabrandi — COMMENTED** [review]
Good work with the tests! I had two comments.

File: cg_lims/EPPs/udf/calculate/calculate_microbial_aliquot_volumes.py (hunk 1 above)
Comment (barrystokman) [reply cg_lims/EPPs/udf/calculate/calculate_microbial_aliquot_volumes.py:23]: I put a reference to the AM-doc in the module docstring here https://github.com/Clinical-Genomics/cg_lims/pull/133/commits/1ea77650104d6c1288dc31f9e6bf5ddcf9135e05

File: cg_lims/EPPs/udf/calculate/calculate_microbial_aliquot_volumes.py (hunk 2 above)
Comment (barrystokman) [reply cg_lims/EPPs/udf/calculate/calculate_microbial_aliquot_volumes.py:65]: raising an exception added here: https://github.com/Clinical-Genomics/cg_lims/pull/133/commits/1ea77650104d6c1288dc31f9e6bf5ddcf9135e05

Comment (mayabrandi) [reply cg_lims/EPPs/udf/calculate/calculate_microbial_aliquot_volumes.py:71]: 👍

File: cg_lims/EPPs/udf/calculate/calculate_microbial_aliquot_volumes.py (hunk 1 above)
Comment (mayabrandi) [reply cg_lims/EPPs/udf/calculate/calculate_microbial_aliquot_volumes.py:25]: 👍

**Review by mayabrandi — APPROVED** [review]

**Comment by barrystokman** [comment]
retest qc flag:
image

image
image
✔️

**Comment by barrystokman** [comment]
Add log file for EPP:

The three file placeholders in the master step are used by the button `3. Generate CSV for Bravo (doc 1564)`: 

image
image

New file place holder for the EPP log:
image

New command is then:
```
bash -c -l "conda activate cg_lims && lims -c /home/glsai/.genologics.yaml epps -l {compoundOutputFileLuid3} -p {processLuid} udf calculate calculate-microbial-aliquot-volumes"
```

**Pull request merged**